Lib/test/test_imp.py

```diff
@@ -213,15 +213,18 @@ def test_load_from_source(self):
         # state after reversion. Reinitialising the module contents
         # and just reverting os.environ to its previous state is an OK
         # workaround
-        orig_path = os.path
-        orig_getenv = os.getenv
+        import types as mod
+        expect_same = ('SimpleNamespace', mod.SimpleNamespace)
+        expect_changed = ('new_class', mod.new_class)
         with os_helper.EnvironmentVarGuard():
-            x = imp.find_module("os")
+            x = imp.find_module("types")
```

Comment:
Running the tests with "-X frozen_modules=on" caused failures.  This was due to either the test expecting a source module or because `os.path` is weird.  (`os.path` looks like a submodule but `os` isn't a package.  Furthermore, `os.path.__name__` isn't "os.path".)  The easiest thing was to switch to a simpler module that isn't frozen, types.

However, after thinking about it I think sticking with `os` (and `os.path`) is a good thing, so I've made adjustments in the tests to make it work.